packages/client/src/modals/TransactionSuccessModal.js

```diff
@@ -0,0 +1,134 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard, useContacts } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab } from "../components/Svg";
+import { getFlowscanUrlForTransaction, formatAddress } from "utils";
+import { ACTION_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const TransactionSuccessModal = ({ actionData, txID, onClose, safeName, safeAddress }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+    const [image, setImage] = useState();
+    const { contacts } = useContacts(safeAddress);
+
+    const { recipient, nftId, collectionId, vaultId, tokenAmount } = actionData;
+    const { displayName, icon } = getTokenMeta(vaultId) || {};
+    const { name: imageName, imageURI } = image || {};
+
+    const actionType = actionData.type;
+    
+    const NFTName = collectionId?.split(".")[2];
+    const NFTAddress = collectionId?.split(".")[1];
+
+    useEffect(() => {
+        if (actionType === ACTION_TYPES.TRANSFER_NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(NFTName, formatAddress(NFTAddress), recipient, nftId);
+                setImage(result);
+            };
+            getImageURL().catch(console.error);
+        }
+    }, [actionType, NFTName, NFTAddress, nftId, recipient, getNFTReference])
+
+    function getTokenMeta(vaultId) {
+        if (vaultId) {
+            const tokenName = vaultId.split(".")[2];
+            return COIN_TYPE_TO_META[CONTRACT_NAME_TO_COIN_TYPE[tokenName]];
+        }
+    }
+
+    function getSafeContactName(address) {
+        const contact = contacts.find(contact => contact.address === address);
+        return contact?.name;
+    }
+
+    return (
+        <div className="p-5 has-text-black has-text-left">
+            <div className="p-5 success-modal-background">
+                <button className="button flex-1 is-info mb-1">
+                    Sent <ArrowUp className="ml-2 has-text-white" />
+                </button>
+                <div className="pl-4">
+                    {actionType === ACTION_TYPES.TRANSFER_NFT &&
+                        <>
+                            <img className="columns is-vcentered is-multiline is-mobile mr-2 mt-2 success-modal-image" src={imageURI} alt={imageName} />
```

Comment:
imageURI could be undefined, suggest checking it first

Reply:
Most probably won't be defined, but if it is, we will display it with an alt.
It's a good question for the team, what should we present in case something is wrong with the image -> some default image from our end or not display image at all?

Reply:
I would say not display as we don't have control on that data.

Reply:
Ok, added logic around that too!